activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -869,6 +869,9 @@ def retryable_query_error?(exception)
           # least raise until the innermost savepoint is rolled back
           @raw_connection&.transaction_status != ::PG::PQTRANS_INERROR &&
             super
+        # If the connection is closed, defer to super
+        rescue PG::ConnectionBad
+          super
```

Comment:
```suggestion
          in_transaction = begin
            @raw_connection&.transaction_status == ::PG::PQTRANS_INERROR
          rescue PG::ConnectionBad
            false
          end
          !in_transaction && super
```

We should scope the `rescue` to `transaction_status` specifically. Because we don't know what super does, so if somehow it raises `rescue PG::ConnectionBad` that will end up in a catastrophic infinite loop.